src/home/homectl.c

```diff
@@ -1246,9 +1246,13 @@ static int create_home(int argc, char *argv[], void *userdata) {
 
                 /* If password quality enforcement is disabled, let's at least warn client side */
 
-                r = user_record_quality_check_password(hr, hr, &error);
-                if (r < 0)
-                        log_warning_errno(r, "Specified password does not pass quality checks (%s), proceeding anyway.", bus_error_message(&error, r));
+                r = user_record_check_password_quality(hr, hr, &error);
+                if (r < 0) {
+                        if (sd_bus_error_has_name(error, BUS_ERROR_LOW_PASSWORD_QUALITY))
+                                log_warning_errno(r, "Specified password does not pass quality checks (%s), proceeding anyway.", bus_error_message(&error, r));
+                        else
+                                return log_error_errno(r, "Failed to check password quality: %m");
```

Comment:
use `bus_error_message() here instead of %m, since if there is a bug error msg set we want to show that